File: frontend/src/app/models/reclamation.ts
export interface Reclamation {
  id: number;
  userId: number;
  eventId?: number;
  reservationId?: string;
  ticketId?: number;
  subject: string;
  description: string;
  type: 'EVENT_ISSUE' | 'RESERVATION_ISSUE' | 'TICKET_ISSUE' | 'PAYMENT_ISSUE' | 'OTHER';
  status: 'OPEN' | 'IN_PROGRESS' | 'RESOLVED' | 'CLOSED' | 'REJECTED';
  priority: 'LOW' | 'MEDIUM' | 'HIGH' | 'CRITICAL';
  assignedTo?: number;
  response?: string;
  resolvedAt?: string;
  createdAt: string;
  updatedAt?: string;
  username?: string;
  eventTitle?: string;
}

export interface ReclamationRequest {
  userId: number;
  eventId?: number;
  reservationId?: string;
  ticketId?: number;
  subject: string;
  description: string;
  type?: string;
}

File: microservices/reclamation-service/src/main/java/com/example/reclamationservice/entity/Reclamation.java
package com.example.reclamationservice.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "reclamations")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Reclamation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private Long userId;

    private Long eventId;

    private String reservationId;

    private Long ticketId;

    @Column(nullable = false)
    private String subject;

    @Column(length = 2000, nullable = false)
    private String description;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ReclamationType type;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    @Builder.Default
    private ReclamationStatus status = ReclamationStatus.OPEN;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    @Builder.Default
    private ReclamationPriority priority = ReclamationPriority.MEDIUM;

    private Long assignedTo;

    @Column(length = 2000)
    private String response;

    private LocalDateTime resolvedAt;

    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}
